boxes.lisp

```diff
@@ -537,3 +540,39 @@ N.B., Successive calls may return the same object."
 
 (defmethod baseline ((box sqrt-box))
   (baseline (sqrt-box-base box)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;; COLUMN BOXES ;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defclass column-box (box)
+  ((padding :initarg :padding
+            :type (integer 0)
+            :accessor column-box-padding)
+   (align :initarg :align
+	  :accessor column-box-align)
+   (contents :initarg :contents
+             :accessor column-box-contents))
+  (:documentation "A vertical concatenation of boxes."))
+
+(defun column-box (boxes &key (padding 0) (align 'left))
+  (make-instance 'column-box :padding padding
+		             :align align
+			     :contents boxes))
+
+(defmethod width ((box column-box))
+  ;; include padding?
+  (maximum (column-box-contents box)
+	   :key #'width))
+
+(defmethod height ((box column-box))
+  (let* ((items (column-box-contents box))
+	 (nb-items (length items)))
+    (+ (sum items :key #'height)
+       (if (zerop nb-items)
+	   0
+	   (* (1- nb-items)
+	      (column-box-padding box))))))
+
+(defmethod baseline ((box column-box))
```

Comment:
I would make the baseline the middle: `(floor (height box) 2)`. Think about something like adding column vectors together. You'd want the centers to align to the `+`'s.